src/clients/c++/CMakeLists.txt

```diff
@@ -26,275 +26,12 @@
-# in the same directory. These source versions are used in the client
-# container install.
-set(TRTIS_CLIENT_HEADER_FLAT 1)
-configure_file(request.h.in flat_include/request.h @ONLY)
-configure_file(request_grpc.h.in flat_include/request_grpc.h @ONLY)
-configure_file(request_http.h.in flat_include/request_http.h @ONLY)
-configure_file(simple_client.cc.in flat_src/simple_client.cc @ONLY)
-
-# Source with full include paths needed for build.
-set(TRTIS_CLIENT_HEADER_FLAT 0)
-configure_file(request.h.in request.h @ONLY)
-configure_file(request_grpc.h.in request_grpc.h @ONLY)
-configure_file(request_http.h.in request_http.h @ONLY)
-configure_file(simple_client.cc.in simple_client.cc @ONLY)
-
-# librequest object build
-set(
-  REQUEST_SRCS
-  request.cc request_common.cc request_http.cc request_grpc.cc
-)
-
-set(
-  REQUEST_HDRS
-  request.h request_common.h
-)
-
-add_library(
-  request-library EXCLUDE_FROM_ALL OBJECT
-  ${REQUEST_SRCS} ${REQUEST_HDRS}
-)
-add_dependencies(
-  request-library
-  model-config-library grpc-library proto-library
-)
-
-# librequest_static.a
-add_library(
-  request_static STATIC
-  $<TARGET_OBJECTS:grpc-library>
-  $<TARGET_OBJECTS:model-config-library>
-  $<TARGET_OBJECTS:proto-library>
-  $<TARGET_OBJECTS:request-library>
-)
-target_link_libraries(
-  request_static
-  PRIVATE gRPC::grpc++
-  PRIVATE gRPC::grpc
-  PUBLIC protobuf::libprotobuf
-  PUBLIC ${CURL_LIBRARY}
-)
-if(NOT WIN32)
-  target_link_libraries(
-    request_static
-    PUBLIC -lpthread
-  )
-endif()
-
-# librequest.so
-add_library(
-  request SHARED
-  $<TARGET_OBJECTS:grpc-library>
-  $<TARGET_OBJECTS:model-config-library>
-  $<TARGET_OBJECTS:proto-library>
-  $<TARGET_OBJECTS:request-library>
-)
-target_include_directories(request PRIVATE ${CURL_INCLUDE_DIRS})
-set_target_properties(
-  request
-  PROPERTIES LINK_DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/librequest.ldscript
-)
-set_target_properties(
-  request
-  PROPERTIES LINK_FLAGS "-Wl,--version-script=librequest.ldscript"
-)
-target_link_libraries(
-  request
-  PRIVATE gRPC::grpc++
-  PRIVATE gRPC::grpc
-  PUBLIC protobuf::libprotobuf
-  PUBLIC ${CURL_LIBRARY}
-)
-if(NOT WIN32)
-  target_link_libraries(
-    request
-    PUBLIC -lpthread
-  )
-endif()
-
-install(
-  TARGETS request
-  LIBRARY DESTINATION lib
-)
-install(
-  TARGETS request_static
-  ARCHIVE DESTINATION lib
-)
-install(
-  FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/flat_include/request.h
-    ${CMAKE_CURRENT_BINARY_DIR}/flat_include/request_grpc.h
-    ${CMAKE_CURRENT_BINARY_DIR}/flat_include/request_http.h
-  DESTINATION include
-)
-install(
-  FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/../../core/api.pb.h
-    ${CMAKE_CURRENT_BINARY_DIR}/../../core/model_config.pb.h
-    ${CMAKE_CURRENT_BINARY_DIR}/../../core/request_status.pb.h
-    ${CMAKE_CURRENT_BINARY_DIR}/../../core/server_status.pb.h
-  DESTINATION include
-)
+add_subdirectory(library)
 
 if(WIN32)
   message("Currently, CMake will only build client library and skip example "
```

Comment:
Move this to example, this message is explaining the examples will not be built for Windows

Reply:
Done